crates/nu-cli/src/repl.rs

```diff
@@ -368,7 +416,14 @@ pub fn evaluate_repl(
         // Right before we start our prompt and take input from the user,
         // fire the "pre_prompt" hook
         if let Some(hook) = config.hooks.pre_prompt.clone() {
-            if let Err(err) = eval_hook(engine_state, stack, None, vec![], &hook, "pre_prompt") {
+            if let Err(err) = eval_hook(
+                engine_state,
+                &mut stack.child_stack(),
```

Comment:
unsure of whether it's right to isolate the stack changes here. Should a hook be able to modify the stack's final state?